### PR TITLE
[MOBILE-1898] Updating Xamarin to Android 14.0.0 and iOS 14.1.1

### PR DESCRIPTION
**Pull request by Apekka**
### What do these changes do?
Updating Xamarin to version 14.0.0, supporting Android SDK 14.0.0 & iOS SDK 14.1.1

### Why are these changes necessary?
To stay up to date

### How did you verify these changes?
Testing sample app

### CODE REVIEW

**Review by marc-scig — COMMENTED** [review]
If the packages are not building for you locally then it's not actually tested. You'll need the changes in the `wcarthage` branch in order to get the carthage build failures to go away. We also need to generate new bindings for the iOS SDK so that any new or changed APIs (such as the URL allow list stuff mentioned in the CHANGELOG) are available.  You can get that process started running the `generateIOSBindings` task in gradle for each of the modules.

**Comment by marc-scig** [comment]
if you haven't already you should also install pyenv and make your global python version 3.x so that xamobni will work

**Comment by marc-scig** [comment]
Packages should be building locally, and iOS changes ready for review now. I think we still need to do some work on Android to support Room (probably need to tweak the Android X package dependencies).

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Airship Xamarin Changelog
 
+## Version 14.0.0 - Sept 28, 2020
+Major release to support iOS SDK 14.1.2 & Android SDK 14.0.1.
+
+### Changes
```
Comment (rlepinski) [inline CHANGELOG.md:6]: We also added some methods on deleting/reading messages and fixes the channel update listener.

Comment (rlepinski) [reply CHANGELOG.md:6]: @oristanovic has docs that need https://urbanairship.atlassian.net/browse/MOBILE-1924 in as well. Looks like a small lift, can we get it in real fast for the release?

**Review by rlepinski — APPROVED** [review]

**Review by rlepinski — APPROVED** [review]

**Pull request merged**